internal/rbd/driver/driver.go

```diff
@@ -125,8 +126,13 @@ func (r *Driver) Run(conf *util.Config) {
 			})
 	}
 
+	nodeLabels, err = k8s.GetNodeLabels(conf.NodeID)
+	if err != nil {
+		log.FatalLogMsg(err.Error())
+	}
```

Comment:
This code will cause problems for non-kubernetes CO. do this check only if the labels are passed?

Reply:
Will move this under 
```go
if conf.EnableReadAffinity {
  ...
}
```

and will make `--enable-ready-affinity` command line flag to be mandatory for enabling read affinity.

Reply:
i was thinking do we really need to have this flag, if the labels are set assume that you will enable read-affinity like we do currently?

Reply:
> This code will cause problems for non-kubernetes CO. do this check only if the labels are passed?

@Madhu-1 we can have a new `--container-orchestrator` flag _(default - kubernetes)_. So we will be fetching node labels only when the flag is set to _kubernetes_?

Reply:
i would vote for new `--container-orchestrator` which might be useful to decide when to make kube API calls. @nixpanic @Rakshith-R thought?

Reply:
> i was thinking do we really need to have this flag, if the labels are set assume that you will enable read-affinity like we do currently?

labels maybe set in the configmap instead. 

Let's have the read-affinity flag set in cmdline as a pre-requisite. 
This setting can be overridden by individual cluster id in configmap.

> i would vote for new `--container-orchestrator` which might be useful to decide when to make kube API calls. @nixpanic @Rakshith-R thought?

I think this entails changes at a lot of places and is out of scope for this pr.

> and will make --enable-ready-affinity command line flag to be mandatory for enabling read affinity.

This option seems sufficient for me.